README.md

```diff
@@ -1 +1,147 @@
-# opencraft
+OpenCraft
+=========
+
+Install
+-------
+
+Instructions based on Ubuntu 14.04.
+
+Install the system package dependencies & virtualenv:
+
+```
+$ sudo apt-get install `cat debian_packages.lst`
+$ pip3 install --user virtualenv && pip3 install --user virtualenvwrapper
+```
+
+Ensure you load virtualenv with Python 3 in `~/.bashrc`:
+
+```
+export PATH="$PATH:$HOME/.local/bin" VIRTUALENVWRAPPER_PYTHON=/usr/bin/python3
+source $HOME/.local/bin/virtualenvwrapper.sh
+```
+
+Then reload `~/.bashrc`, create the virtual env and install the Python requirements:
+
+```
+$ . ~/.bashrc
+$ mkvirtualenv -p /usr/bin/python3 opencraft
+$ pip install -r requirements.txt
+```
+
+
+Configure
+---------
+
+Create an `.env` file at the root of the repository or set environment variables, customizing the
+settings from `opencraft/settings.py` which are loaded via `env()`.
+
+
+Run
+---
+
+To run the development server:
+
+```
+$ make rundev
+```
+
+Then go to:
+
+* User interface: [http://localhost:5000/](http://localhost:2000/)
+* API: [http://localhost:5000/api/](http://localhost:2000/api/)
+* Admin: [http://localhost:5000/admin/](http://localhost:2000/admin/)
+
+To run the production server:
+
+```
+$ make run
+```
+
+To change the number of concurrent workers ran by the production server:
+
+```
+$ make run WORKERS=2
+```
+
+
+Processus description
+---------------------
+
+This runs three processus via honcho, which reads `Procfile` or `Procfile.dev` and loads the
+environment from the `.env` file:
+
+* *web*: the main HTTP server (Django - Werkzeug debugger in dev, gunicorn in prod)
+* *websocket*: the websocket server (Tornado)
+* *worker*: runs asynchronous jobs (Huey)
+
+Important: the Werkzeug debugger started by the development server allows remote execution
+of Python commands. It should *not* be run in production.
+
+
+Static assets collection
+------------------------
+
+The Web server started in the development environment also doesn't require to run collectstatic
+after each change.
+
+The production environment automatically runs collectstatic on startup, but you can also run it
+manually:
+
+```
+$ make collectstatic
+```
+
+
+Migrations
+----------
+
+Similarly, migrations are run automatically on startup - for both the development and production
+environments. To run it manually:
+
+```
+$ make migrate
+```
+
+
+Running the tests
+-----------------
+
+First, ensure that the postgresql user can create databases, to be able to create the test database. 
+Then run the whole test suite (pylint, pyflakes, pep8, unit tests, etc.) with:
+
+```
+$ make test
+```
+
+To run a single test, use `make test_one`:
+
+```
+$ make testone instance.tests.models.test_server
```

Comment:
It used to be `testone`, but changed to `test_one` without updating the README - thanks for catching this, updated.